src/Bundler/Runner/UnixSocketRunner.php

```diff
@@ -0,0 +1,133 @@
+                usleep(500000);
+                continue;
+            }
+
+            if (!@socket_connect($socket, '/tmp/build')) {
+                usleep(100000);
+                continue;
+            }
+
+            try {
+                $response = $this->sendMessage($socket, $type, $file_name, $item->getContent());
+                break;
+            } catch (SocketException $e) {
+                usleep(100000);
+                continue;
+            }
+        }
+
+        socket_close($socket);
+
+        return $response;
+    }
+
+    private function sendMessage($socket, $type, $file_name, $msg)
+    {
+        /*
+         * The request header is 7 bytes:
+         * - 4 bytes File length, unsigned long (32 bit, little endian byte order)
+         * - 3 bytes for a three character system name of the operation to perform
+         *
+         * After that the file is sent.
+         */
+        if (strlen($type) !== 3) {
+            throw new \Exception('Type is always three bytes');
+        }
+        $msg_length = strlen($msg);
+
+        $to_send = pack('V', $msg_length) . $type . $file_name . "\0" . $msg;
+        $length  = 4 + 3 + strlen($file_name) + 1 + $msg_length;
+        if (false === @socket_send($socket, $to_send, $length, 0)) {
+            throw new SocketException('Problem sending to the socket');
+        }
+
+        /**
+         * The response header is
+         * - 4 bytes Resulting file length, unsigned long (32 bit, little endian byte order)
+         * - 1 byte boolean flags
+         *
+         * $flags & 1 => Success
+         * $flags & 2 => I just killed myself
+         * (The other 6 flags left open)
+         *
+         * After that the result is sent.
+         */
+        $buffer = null;
+        if (false === @socket_recv($socket, $buffer, 4, MSG_WAITALL)) {
+            throw new SocketException('Problem receiving from the socket');
+        }
+        $length = unpack('Vlength', $buffer)['length'];
+        if (false === @socket_recv($socket, $buffer, 1, MSG_WAITALL)) {
+            throw new SocketException('Problem receiving from the socket');
+        }
+        $flags   = ord($buffer);
+        $success = $flags & 1;
+        $restart = $flags & 2;
+
+        if (false === @socket_recv($socket, $buffer, $length, MSG_WAITALL)) {
+            throw new SocketException('Problem receiving from the socket');
+        }
+
+        if ($success) {
+            return $buffer;
+        }
+
+        if ($restart) {
+            $this->startBuildProcess();
+        }
+
+        throw new \Exception(sprintf('Error with %s compile: %s', $type, $buffer));
+    }
+
+    private function startBuildProcess()
+    {
+        $node_js = $this->config->getNodeJsExecutable();
+        $cmd     = sprintf(
+            'nohup %s %s > /dev/null &',
+            $node_js->getBinary(),
+            implode(DIRECTORY_SEPARATOR, [__DIR__, '..', '..', 'Resources', 'build.js'])
+        );
+        `$cmd`;
```

Comment:
possible usecase for this `symfony/process` component? It is already used for the other runner.